### PR TITLE
script to generate chapters

### PR DESCRIPTION
**Pull request by gowdan**

### CODE REVIEW

File: scripts/gen_chapters.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  map_config = require(map_config_path),
+  dat = fs.readFileSync(map_config_path, 'utf8'),
+  map_data = {},
+  map_id = process.argv[2],
+  width = map_config.width,
+  tileWidth = map_config.tileWidth,
+  tileHeight = map_config.tileHeight,
+  mapLength = map_config.maps[0].length,
+  mapRepeat = map_config.maps[0].repeat,
+  bridgeLength = map_config.bridge.length,
```
Comment (rampr) [inline scripts/gen_chapters.js]: some variables are using camel case and some have underscores in them. Let's follow one style

File: scripts/gen_chapters.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  last_chapter_height = map_config.height,
+  first_chapter_height =mapLength*mapRepeat+bridgeLength,
```
Comment (rampr) [inline scripts/gen_chapters.js]: space between operators would make it more readable

Comment (gowdan) [reply scripts/gen_chapters.js]: ok  I will change

File: scripts/gen_chapters.js (hunk 1 above)
Comment (gowdan) [reply scripts/gen_chapters.js]: ok I will follow underscores

File: scripts/gen_chapters.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  total_chapters = map_config.length,
+  data = fs.readFileSync('maps/final_map.js', 'utf8'),
+  chapters = [],
+  id = 1,
+  grid;
+
+for (var i = map_config.height; i >= first_chapter_height; i-=first_chapter_height) {
+  chapters.push(i);
+}
+chapters = _.map(chapters, function (val) {return val - 3;});
```
Comment (rampr) [inline scripts/gen_chapters.js]: why can't we do this in the previous for loop itself ?
Also add a comment on why you are subtracting by 3.

Will be difficult to understand in the future why there is a magic number 3

File: scripts/gen_chapters.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+data = JSON.parse(data.split('exports = ')[1].split(';')[0]);
+
```
Comment (rampr) [inline scripts/gen_chapters.js]: newline is unnecessary

File: scripts/gen_chapters.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+grid = _.reverse(data.grid);
```
Comment (rampr) [inline scripts/gen_chapters.js]: a new line here would make it better to read

**Comment by rampr** [comment]
Add meaningful commit messages

File: scripts/gen_chapters.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+_.each(grid, function (row, idx) {
+  if (chapters.indexOf(idx) > -1) {
+    var tile = row[0];
+
+    row[0] = {
+      chapter: true,
+      map: tile,
+      id: id
+    };
+
```
Comment (rampr) [inline scripts/gen_chapters.js:40]: unnecesary new line

File: scripts/gen_chapters.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+/* jshint node:true */
```
Comment (vivekb91) [inline scripts/gen_chapters.js:1]: We have read me for scripts. please add this scripts info also there

File: scripts/gen_chapters.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+"use strict";
+
+var _ = require('lodash'), 
+  path = require('path'),
+  fs = require('fs'),
+  map_config_path = process.argv[4] ? process.argv[4] :
+  path.join(__dirname, 'map_config.json'),
+  map_config = require(map_config_path),
+  dat = fs.readFileSync(map_config_path, 'utf8'),
```
Comment (Javasamurai) [inline scripts/gen_chapters.js]: Are we using this?

Comment (gowdan) [reply scripts/gen_chapters.js]: yes we are, for getting the length of map and how many times it is repeating.

File: readme_scripts.md [hunk 9]
```diff
@@ -5,6 +5,8 @@ Scripts:
 
     3. map_config - This has config values like width, height of the map. Comingsoon shows the length of comingsoon and bridge. Maps object is used to define the length of each map and how many times it's repeated.
 
+    4. gen_chapters - Adds the chapters based on the values from map_config script like length of maps and repetations of each maps. And it will export added chapters to maps/final_map.js.
```
Comment (rampr) [inline readme_scripts.md]: typo in repetitions.

File: scripts/gen_chapters.js (hunk 6 above)
Comment (gowdan) [reply scripts/gen_chapters.js:40]: removed

File: scripts/gen_chapters.js (hunk 7 above)
Comment (gowdan) [reply scripts/gen_chapters.js:1]: ok I have added

File: readme_scripts.md (hunk 9 above)
Comment (gowdan) [reply readme_scripts.md]: ok changed

**Review by rampr — APPROVED** [review]

**Review by vivekb91 — APPROVED** [review]

**Comment by vivekb91** [comment]
@gowdan We can merge this. update package before merging and create after.

**Review by Javasamurai — APPROVED** [review]

**Comment by Javasamurai** [comment]
### **Issues:**
lodash is not there in the dependencies add it in package.json
script adds chapters in last milestone also which is not required.

**Pull request merged**